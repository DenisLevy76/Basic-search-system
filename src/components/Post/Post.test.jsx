import { render, screen } from "@testing-library/react";
import { Post } from ".";

const mockPost = {
  id: 1,
  title: "Hello, world!",
  body: "Todos os tipos de hello, world",
  img: "https://github.com/DenisLevy76.png",
};

describe("<Post />", () => {
  test("It should render a post", () => {
    render(<Post post={mockPost} />);

    //um preview de como o html fica dps de renderizar
    // const {debug} = render(<Post post={mockPost}/>)

    // debug();
  });

  test("It should contais a img", () => {
    render(<Post post={mockPost} />);

    const img = screen.getByAltText(mockPost.title);

    expect(img).toHaveAttribute("src", mockPost.img);
  });

  test("It should contais a h2 with post title", () => {
    render(<Post post={mockPost} />);

    const h2 = screen.getByRole("heading", {
      name: mockPost.title + " " + mockPost.id,
    });

    expect(h2).toBeInTheDocument();
  });

  test("It should contais a p with post body", () => {
    render(<Post post={mockPost} />);

    const p = screen.getByText(mockPost.body);

    expect(p).toBeInTheDocument();
  });

  test("It should match snapshot", () => {
    const { container } = render(<Post post={mockPost} />);
    expect(container).toMatchSnapshot();
  });
});
