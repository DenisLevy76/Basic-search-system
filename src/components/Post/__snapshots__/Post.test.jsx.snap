// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> It should match snapshot 1`] = `
<div>
  <div
    class="post-container"
  >
    <img
      alt="Hello, world!"
      src="https://github.com/DenisLevy76.png"
    />
    <div
      class="post-body"
    >
      <h2>
        Hello, world!
         
        1
      </h2>
      <p>
        Todos os tipos de hello, world
      </p>
    </div>
  </div>
</div>
`;
